File: Arrays_Lists/questions_py/question_1.1.py
def sum_of_pairs(arr, n):
    for i in range(len(arr) // 2):
        for j in range(i + 1, len(arr)):
            if arr[i] + arr[j] == n:
                print([arr[i], arr[j]])
sum_of_pairs([1, 2, 3, 4, 5, 6, 7, 8, 9, 10], 10)
# sum_of_pairs([2, 3, 6, 9], 9)

# best solution
def two_sum(nums, target):
    seen = {}
    
    for i, num in enumerate(nums):
        complement = target - num
        
        if complement in seen:
            return [seen[complement], i]
        
        seen[num] = i
 
nums = [2, 7, 11, 15]
target = 9
indices = two_sum(nums, target)
print(f"Indices of the two numbers are: {indices}")

File: Arrays_Lists/questions_c/question_1_1.c
#include <stdio.h>

void sum_pairs(int arr[], int n);

int main() {
    int nums[] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, n = 10, result;
    sum_pairs(nums, n);
    return 0;
}

void sum_pairs(int arr[], int n) {
    for(int i = 0; i < (n/2); i++) {
        for(int j = i + 1; j < n; j++) {
            if ((arr[i] + arr[j]) == n) {
                printf("(%d, %d)\n", arr[i], arr[j]);
            }
        }
    }
}